tests/unit/withOnyxTest.js

```diff
@@ -153,3 +153,70 @@ describe('withOnyx', () => {
+        const onRender1 = jest.fn();
+        const onRender2 = jest.fn();
+        const onRender3 = jest.fn();
+
+        // GIVEN there is a collection with three simple items in it
+        Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+            test_1: {ID: 1},
+            test_2: {ID: 2},
+            test_3: {ID: 3},
+        });
+
+        return waitForPromisesToResolve()
+            .then(() => {
+                // WHEN three components subscribe to each of the items in that collection
+                const TestComponentWithOnyx1 = compose(
+                    withOnyx({
+                        testObject: {
+                            key: `${ONYX_KEYS.COLLECTION.TEST_KEY}1`,
+                        },
+                    }),
+                )(ViewWithCollections);
+                render(<TestComponentWithOnyx1 onRender={onRender1} />);
+
+                const TestComponentWithOnyx2 = compose(
+                    withOnyx({
+                        testObject: {
+                            key: `${ONYX_KEYS.COLLECTION.TEST_KEY}2`,
+                        },
+                    }),
+                )(ViewWithCollections);
+                render(<TestComponentWithOnyx2 onRender={onRender2} />);
+
+                const TestComponentWithOnyx3 = compose(
+                    withOnyx({
+                        testObject: {
+                            key: `${ONYX_KEYS.COLLECTION.TEST_KEY}3`,
+                        },
+                    }),
+                )(ViewWithCollections);
+                render(<TestComponentWithOnyx3 onRender={onRender3} />);
+
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                // WHEN a single item in the collection is updated with mergeCollect()
+                Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+                    test_1: {ID: 1, newProperty: 'yay'},
+                });
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                // THEN the component subscribed to the modified item should have the new version of the item
+                // and all other components should be unchanged.
+                // Note: each component is rendered twice. Once when it is initially rendered, and then again
+                // when the collection is updated. That's why there are two checks here for each component.
+                expect(onRender1.mock.calls[0][0].testObject).toStrictEqual({ID: 1});
+                expect(onRender1.mock.calls[1][0].testObject).toStrictEqual({ID: 1, newProperty: 'yay'});
+                expect(onRender2.mock.calls[0][0].testObject).toStrictEqual({ID: 2});
+                expect(onRender2.mock.calls[1][0].testObject).toStrictEqual({ID: 2});
+                expect(onRender3.mock.calls[0][0].testObject).toStrictEqual({ID: 3});
+                expect(onRender3.mock.calls[1][0].testObject).toStrictEqual({ID: 3});
```

Comment:
Like my expectation would be... not that this gets called again with same data but not called at all ? Why should it re-render since nothing has changed?